integration-tests/prepare-operation.spec.ts

```diff
@@ -0,0 +1,138 @@
+      done();
+    });
+
+    it('should be able to prepare a transaction operation', async (done) => {
+      const prepared = await Tezos.prepare.transaction({
+        to: 'tz1QZ6KY7d3BuZDT1d19dUxoQrtFPN2QJ3hn',
+        amount: 5
+      });
+
+      expect(prepared).toBeDefined();
+      expect(prepared.counter).toBeDefined();
+      expect(prepared.opOb).toBeDefined();
+      expect(prepared.opOb.branch).toBeDefined();
+      expect(prepared.opOb.contents).toBeDefined();
+
+      const content = prepared.opOb.contents[0] as OperationContentsTransaction;
+
+      expect(content.kind).toEqual('transaction');
+      expect(content.destination).toEqual('tz1QZ6KY7d3BuZDT1d19dUxoQrtFPN2QJ3hn')
+      expect(content.amount).toEqual('5000000');
+
+      done();
+    });
+
+    it('should be able to prepare a batch operation', async (done) => {
+      const prepared = await Tezos.prepare.batch([
+        {
+          kind: OpKind.TRANSACTION,
+          to: 'tz1QZ6KY7d3BuZDT1d19dUxoQrtFPN2QJ3hn',
+          amount: 2,
+        },
+        {
+          kind: OpKind.TRANSACTION,
+          to: 'tz1QZ6KY7d3BuZDT1d19dUxoQrtFPN2QJ3hn',
+          amount: 2,
+        },
+      ]);
+
+      expect(prepared).toBeDefined();
+      expect(prepared.counter).toBeDefined();
+      expect(prepared.opOb).toBeDefined();
+      expect(prepared.opOb.branch).toBeDefined();
+      expect(prepared.opOb.contents).toBeDefined();
+      expect(prepared.opOb.contents.length).toEqual(2);
+      expect(prepared.opOb.contents[0].kind).toEqual('transaction');
+      expect(prepared.opOb.contents[1].kind).toEqual('transaction');
+
+      done();
+    });
+
+    it('should be able to prepare a txRollupOriginate operation', async (done) => {
+      const prepared = await Tezos.prepare.txRollupOrigination({});
+
+      expect(prepared).toBeDefined();
+      expect(prepared.counter).toBeDefined();
+      expect(prepared.opOb).toBeDefined();
+      expect(prepared.opOb.branch).toBeDefined();
+      expect(prepared.opOb.contents).toBeDefined();
+      expect(prepared.opOb.contents[0].kind).toEqual('tx_rollup_origination');
+
+      done();
+    });
+
+    it('should be able to prepare a ballot operation', async (done) => {
+      const prepared = await Tezos.prepare.ballot({
+        proposal: 'PtKathmankSpLLDALzWw7CGD2j2MtyveTwboEYokqUCP4a1LxMg',
+        ballot: 'yay'
+      });
+
+      expect(prepared).toBeDefined();
+      expect(prepared.counter).toBeDefined();
+      expect(prepared.opOb).toBeDefined();
+      expect(prepared.opOb.branch).toBeDefined();
+      expect(prepared.opOb.contents).toBeDefined();
+
+      const content = prepared.opOb.contents[0] as OperationContentsBallot
+
+      expect(prepared.opOb.contents[0].kind).toEqual('ballot');
+      expect(content.proposal).toBeDefined();
+      expect(content.proposal).toEqual('PtKathmankSpLLDALzWw7CGD2j2MtyveTwboEYokqUCP4a1LxMg');
+      expect(content.ballot).toBeDefined();
+      expect(content.ballot).toEqual('yay');
```

Comment:
Something small, if we are expecting proposal and ballot to be 'PtKathmankSpLLDALzWw7CGD2j2MtyveTwboEYokqUCP4a1LxMg', and 'yay' maybe checking if it's `.toBeDefined()` is not needed.

Reply:
good point, I'll remove